### DISCUSSION TITLE
is it possible for my sound bot to play a sound whenever someone gets a specific role?

### DISCUSSION THREAD

**Discussion by ghost** [Rapptz/discord.py#7102] [Q&A]
hi so i have a bot that when someone gets in a vc, it says "welcome to the server". now i want it to play a sound when a user reacts to an emoji for a role? is this possible?  this is the code so far 

```py
import asyncio
import discord
import audioread

#BOT TOKEN
TOKEN = "BOTTOKENHERE" # BOT TOKEN GOES HERE

music_path = r""

client = discord.Client()

@client.event
async def on_ready():
    print(f'{client.user} has connected to Discord!')

@client.event
async def on_voice_state_update(member, before, after):
    #VOICE CHANNEL ID
    VOICE_CHANNEL_ID = CHANNELIDHERE # CHANNEL ID GOES HERE
    path = r"C:\Users\Toby\Desktop\Sound Bot Test\audio.mp3" # PATH OF AUDIO FILE GOES HERE, EXAMPLE: C:\Users\User\Desktop\Sound Bot Test\audio.mp3
    vc_before = before.channel
    vc_after = after.channel

    if not vc_before and vc_after.id == VOICE_CHANNEL_ID:
        vc = await vc_after.connect()
        vc.play(discord.FFmpegPCMAudio(path))
        with audioread.audio_open(path) as f:
            await asyncio.sleep(f.duration)
        await vc.disconnect()


client.run(TOKEN)```

**Comment by ProManGaming** [comment]
> hi so i have a bot that when someone gets in a vc, it says "welcome to the server". now i want it to play a sound when a user reacts to an emoji for a role? is this possible?  this is the code so far 
> 
> ```py
> import asyncio
> import discord
> import audioread
> 
> #BOT TOKEN
> TOKEN = "BOTTOKENHERE" # BOT TOKEN GOES HERE
> 
> music_path = r""
> 
> client = discord.Client()
> 
> @client.event
> async def on_ready():
>     print(f'{client.user} has connected to Discord!')
> 
> @client.event
> async def on_voice_state_update(member, before, after):
>     #VOICE CHANNEL ID
>     VOICE_CHANNEL_ID = CHANNELIDHERE # CHANNEL ID GOES HERE
>     path = r"C:\Users\Toby\Desktop\Sound Bot Test\audio.mp3" # PATH OF AUDIO FILE GOES HERE, EXAMPLE: C:\Users\User\Desktop\Sound Bot Test\audio.mp3
>     vc_before = before.channel
>     vc_after = after.channel
> 
>     if not vc_before and vc_after.id == VOICE_CHANNEL_ID:
>         vc = await vc_after.connect()
>         vc.play(discord.FFmpegPCMAudio(path))
>         with audioread.audio_open(path) as f:
>             await asyncio.sleep(f.duration)
>         await vc.disconnect()
> 
> 
> client.run(TOKEN)```

Nice

**Comment by izxxr** [comment]
Use [`on_member_update`](https://discordpy.readthedocs.io/en/latest/api.html#discord.on_member_update) event and see for role changes.

**Reply by ghost** [reply]
my friend told me that the bot will crash if i do this. on_member_update doesnt specifically check for roles

**Reply by Rapptz** [reply]
Your friend was mistaken or there was a misunderstanding. `on_member_update` triggers on multiple scenarios, one of which is checking when their roles change. So doing something like 
```py
@bot.event
def on_member_update(before, after):
    if before.roles != after.roles:
        # roles changed
```
will work just fine.

**Reply by asdfchris** [reply]
> Your friend was mistaken or there was a misunderstanding. `on_member_update` triggers on multiple scenarios, one of which is checking when their roles change. So doing something like
> 
> ```python
> @bot.event
> def on_member_update(before, after):
>     if before.roles != after.roles:
>         # roles changed
> ```
> 
> will work just fine.

oh cool I was also under the assumption that on_member_update would trigger all the scenarios listed in the doc. I'm trying to get a similar bot working and am wondering, how could I make the bot join and play a sound when somebody is given a specific role?

**Reply by ghost** [reply]
> Your friend was mistaken or there was a misunderstanding. `on_member_update` triggers on multiple scenarios, one of which is checking when their roles change. So doing something like
> 
> ```python
> @bot.event
> def on_member_update(before, after):
>     if before.roles != after.roles:
>         # roles changed
> ```
> 
> will work just fine.

hmm what if i want it to be a specific role? will it be like:

```py
@bot.event
def on_member_update(before, after):
    if before.roles != after.roles:
             after.roles(role_id):
             path = 
        # roles changed
